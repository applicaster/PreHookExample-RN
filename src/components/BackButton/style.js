import {
  StyleSheet,
} from 'react-native';

const backButtonSize = 25;

export const styles = StyleSheet.create({
  backButton: {
    tintColor: '#FFFFFF',
    height: backButtonSize,
    width: backButtonSize,
    marginLeft: 10,
  },
});
