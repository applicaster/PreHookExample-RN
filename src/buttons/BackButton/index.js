import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  TouchableWithoutFeedback,
} from 'react-native';
import { styles } from './style';

const BACK_BUTTON = 'feed_rn_back_button';
class BackButton extends Component {
  render() {
    const { onPress } = this.props;
    const { textColor = '#FFFFFF' } = this.context.styles;
    const backButtonColor = { tintColor: `${textColor}99` };

    return (
      <TouchableWithoutFeedback onPress={onPress}>
        <Image style={[styles.backButton, backButtonColor]} source={{ uri: BACK_BUTTON }} />
      </TouchableWithoutFeedback>
    );
  }
}

BackButton.propTypes = {
  onPress: PropTypes.func,
};

BackButton.contextTypes = {
  styles: PropTypes.object,
};

export default BackButton;
