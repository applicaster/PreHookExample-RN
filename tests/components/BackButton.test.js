import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BackButton from '../../src/components/BackButton';

describe('BackButton', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <BackButton onPress={() => {}} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
