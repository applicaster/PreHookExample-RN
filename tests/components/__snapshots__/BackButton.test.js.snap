// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component renders correctly 1`] = `
<Image
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  source={
    Object {
      "uri": "feed_rn_back_button",
    }
  }
  style={
    Array [
      Object {
        "height": 25,
        "marginLeft": 5,
        "tintColor": "#FFFFFF",
        "width": 25,
      },
      Object {
        "tintColor": "#FFFFFF99",
      },
    ]
  }
  testID={undefined}
/>
`;
